File: Programmers/Level1/모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int [] a = {1,2,3,4,5};
        int [] b = {2,1,2,3,2,4,2,5};
        int [] c = {3,3,1,1,2,2,4,4,5,5};
        ArrayList<Integer> list = new ArrayList<Integer>();
        int aCount=0, bCount=0, cCount=0;
        
        for(int i=0; i<answers.length; i++){
            if(a[i%a.length]==answers[i]){
                aCount++;
            }
            if(b[i%b.length]==answers[i]){
                bCount++;
            }
            if(c[i%c.length]==answers[i]){
                cCount++;
            }
        }
        int max = Math.max(aCount,Math.max(bCount,cCount));
        if(max==aCount){
            list.add(1);
        }
        if(max==bCount){
            list.add(2);
        }
        if(max==cCount){
            list.add(3);
        }
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++){
            answer[i]=list.get(i);
        }
        return answer;
    }
}

File: Programmers/Level1/모의고사/모의고사.py
def solution(answers):
    answer = []
    
    one = [1,2,3,4,5]
    two = [2,1,2,3,2,4,2,5]
    three = [3,3,1,1,2,2,4,4,5,5]
    tmp = [check(answers, one), check(answers, two), check(answers, three)]
    
    for i,k in enumerate(tmp):
        if k == max(check(answers, one), check(answers, two), check(answers, three)):
            answer.append(i+1)
        
    return answer

def check(answers, nums):
    i = 0
    k = 0
    total_sum = 0

    while i < len(answers):
        if answers[i] == nums[k]: 
            total_sum += 1
            
        k += 1
        
        if k == len(nums):
            k = 0
            
        i += 1
    
    return total_sum
